### PR TITLE
ft: switch to node-rdkafka for producer/consumer

### PR DESCRIPTION
**Pull request by jonathan-gramain**
- Better handling of connections, node failures etc.
- remove CONSUMER_FETCH_MAX_BYTES (now only advisory)
- re-enable and rework BackbeatConsumer tests

Review only until we agree on which release we can get this in.

### CODE REVIEW

File: tests/functional/BackbeatConsumer.js [hunk 1]
```diff
@@ -27,30 +26,53 @@ describe.skip('BackbeatConsumer', () => {
...
+    it('should be able to read messages sent to the topic', done => {
+        let consumeCb = null;
+        let totalConsumed = 0;
+        async.series([
+            next => {
+                consumer.subscribe();
+                consumer.on('consumed', messagesConsumed => {
+                    totalConsumed += messagesConsumed;
+                    assert(totalConsumed <= messages.length);
+                    if (totalConsumed === messages.length) {
```
Comment (philipyoo) [inline tests/functional/BackbeatConsumer.js]: If the assert above passes, will we ever enter this `if` statement block?

Comment (rahulreddy) [reply tests/functional/BackbeatConsumer.js]: It does. The assert doesn't throw an assertion when the condition is true. You can try this in your terminal.
```js
assert(5 <= 5);
if (5 === 5) {
   console.log('hello');
}
```

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by philipyoo** [comment]
@ironman-machine try

**Comment by philipyoo** [comment]
@ironman-machine try

**Comment by philipyoo** [comment]
@ironman-machine try

**Comment by philipyoo** [comment]
@ironman-machine try

**Comment by jonathan-gramain** [comment]
replaced by #201

**Pull request closed without merging**